packages/platform-ios/src/config/index.ts

```diff
@@ -48,13 +48,18 @@ export function projectConfig(folder: string, userConfig: IOSProjectParams) {
   }
 
   const projectPath = path.join(folder, project);
-  const sourceDir = path.dirname(projectPath);
+  const podfile = findPodfilePath(folder, projectPath);
+
+  // This is a temporary fix for #1435. In certain repos, the Xcode project can
+  // be generated by a tool. The only file that we can assume to exist on disk
+  // is `Podfile`.
+  const sourceDir = podfile ? path.dirname(podfile) : path.dirname(projectPath);
```

Comment:
I would not change this one to be honest, `sourceDir` is not needed for `autolinking`, so there's no need to adjust it for cases when `xcodeproj` doesn't exist. It is needed for legacy `link` command and in the case of a missing `xcodeproj`, that would fail anyway.

Reply:
It looks like it _is_ being used for autolinking:
https://github.com/react-native-community/cli/blob/44f541332ee57e8a30feb3ca24479ab292149dd9/packages/platform-ios/native_modules.rb#L40

Without this change, we hit errors similar to this when running `pod install`:
```
[!] No podspec found for `ReactTestApp-DevSupport` in `..`
```

Reply:
You’re right 😅